src/modules/swap/containers/SurplusModalSetup/index.tsx

```diff
@@ -0,0 +1,38 @@
+import { useCallback } from 'react'
+
+import { useOrder } from 'legacy/state/orders/hooks'
+
+import { useWalletInfo } from 'modules/wallet'
+
+import { GpModal } from 'common/pure/Modal'
+import * as styledEl from 'common/pure/TransactionSubmittedContent/styled'
+import { SurplusModal } from 'common/pure/TransactionSubmittedContent/SurplusModal'
+
+import { useOrderIdForSurplusModal, useRemoveOrderFromSurplusQueue } from '../../state/surplusModal'
+
+export function SurplusModalSetup() {
+  const orderId = useOrderIdForSurplusModal()
+  const removeOrderId = useRemoveOrderFromSurplusQueue()
+
+  const { chainId } = useWalletInfo()
+  const order = useOrder({ id: orderId, chainId })
+
+  const onDismiss = useCallback(() => {
+    orderId && removeOrderId(orderId)
+  }, [orderId, removeOrderId])
+
+  const isOpen = !!orderId
+
+  return (
+    <GpModal isOpen={isOpen} onDismiss={onDismiss} maxHeight={90} maxWidth={470}>
+      <styledEl.Wrapper>
+        <styledEl.Section>
+          <styledEl.Header>
+            <styledEl.CloseIconWrapper onClick={onDismiss} />
+          </styledEl.Header>
+        </styledEl.Section>
+        <SurplusModal order={order} />
+      </styledEl.Wrapper>
```

Comment:
Re-using the same components to works nicely with https://github.com/cowprotocol/cowswap/pull/2767 without any changes